### PR TITLE
validate links in docs

### PR DESCRIPTION
**Pull request by pjfanning**
job is failing due to bad scaladoc links for pekko-connectors - package name is partially repeated in the links

```
## HTTP failure response
`[https://pekko.apache.org/api/pekko-connectors/1.0.0/org/apache/pekko/stream/connectors/org/apache/pekko/stream/connectors/cassandra/javadsl/CassandraSession.html`](https://pekko.apache.org/api/pekko-connectors/1.0.0/org/apache/pekko/stream/connectors/org/apache/pekko/stream/connectors/cassandra/javadsl/CassandraSession.html%60) status 404 Not Found
 - ../../../../../target/scala-2.13/unidoc/org/apache/pekko/persistence/cassandra/query/javadsl/CassandraReadJournal.html

`[https://pekko.apache.org/api/pekko-connectors/1.0.0/org/apache/pekko/stream/connectors/org/apache/pekko/stream/connectors/cassandra/scaladsl/CassandraSession.html`](https://pekko.apache.org/api/pekko-connectors/1.0.0/org/apache/pekko/stream/connectors/org/apache/pekko/stream/connectors/cassandra/scaladsl/CassandraSession.html%60) status 404 Not Found
 - ../../../../../target/scala-2.13/unidoc/org/apache/pekko/persistence/cassandra/EventsByTagMigration.html
 - ../../../../../target/scala-2.13/unidoc/org/apache/pekko/persistence/cassandra/query/scaladsl/CassandraReadJournal.html
```

### CODE REVIEW

**Review by nvollmar — APPROVED** [review]
lgtm

**Comment by pjfanning** [comment]
@mdedetrich not urgent but I've run into an issue with this PR. Some of the scaladoc is generating with invalid links (see the description of this PR above). Any ideas? I might have missed something but the build.sbt setup seems correct and that we could be hitting a scaladoc bug. We've seen a couple of those with Pekko builds already.

**Comment by mdedetrich** [comment]
Ill look into it tomorrow

**Comment by mdedetrich** [comment]
Looking at this now

File: scripts/link-validator.conf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// config for https://github.com/ennru/site-link-validator/
+site-link-validator {
+  root-dir = "docs/target/paradox/site/main/"
+  # relative to `root-dir`
+  start-file = "index.html"
+
+  # Resolves URLs with the given prefix as local files instead
+  link-mappings = [
+    {
+      prefix = "https://pekko.apache.org/docs/pekko-persistence-cassandra/1.0.0/"
```
Comment (mdedetrich) [inline scripts/link-validator.conf:12]: So I don't think you are meant to be using 1.0.0 as a prefix here, only current i.e. https://github.com/apache/incubator-pekko/blob/442f54f6fe1da6857b96927c8bc42920b02e3aa0/scripts/link-validator.conf#L8-L39 as an example

**Comment by mdedetrich** [comment]
@pjfanning Found core issue, please see https://github.com/apache/incubator-pekko-connectors/pull/252 . Unfortunately fixing this properly requires a new patch release of pekko-connectors as well as a new RC for this project (since the scaladoc artifacts are part of the maven release, its not just paradox).

**Comment by pjfanning** [comment]
> @pjfanning Found core issue, please see [apache/incubator-pekko-connectors#252](https://github.com/apache/incubator-pekko-connectors/pull/252) . Unfortunately fixing this properly requires a new patch release of pekko-connectors as well as a new RC for this project (since the scaladoc artifacts are part of the maven release, its not just paradox).

Thanks for looking - good catch. I don't think it is urgent to get a 1.0.1 release of pekko-connectors out. Maybe we can wait a few weeks to see if there are more issues. It might be nice to include a few dependency upgrades in a 1.0.1 release.

File: scripts/link-validator.conf (hunk 1 above)
Comment (pjfanning) [reply scripts/link-validator.conf:12]: We use the 1.0.0 trick in other builds - just not in incubator-pekko.

**Comment by mdedetrich** [comment]
> Thanks for looking - good catch. I don't think it is urgent to get a 1.0.1 release of pekko-connectors out. Maybe we can wait a few weeks to see if there are more issues. It might be nice to include a few dependency upgrades in a 1.0.1 release.

Maybe delaying pekko-persistence-cassandra release is a bit much, but I do think we should release 1.0.1 connectors as soon as it can.

Any dependencies (including ours) will have broken doc links and it also means that link validators that happen to use Pekko connectors won't ever pass.

We can always do a 1.0.2 with updates to dependencies, but I would rather not bundle that with the doc fix.

**Comment by pjfanning** [comment]
I don't think we should cancel the persistence-cassandra 1.0.0 vote/release. It is an unfortunate issue that the some links are broken in our docs but I don't think this is really all that bad.

We can start a discussion about a pekko-connectors 1.0.1 release but I would appreciate not bringing up cancelling the persistence-cassandra 1.0.0 vote over this.

**Comment by mdedetrich** [comment]
> We can start a discussion about a pekko-connectors 1.0.1 release but I would appreciate not bringing up cancelling the persistence-cassandra 1.0.0 vote over this.

Sure, I'll make a thread only about pekko-connectors release

**Comment by pjfanning** [comment]
@mdedetrich could you re-review this?

**Review by mdedetrich — APPROVED** [review]
lgtm

**Pull request merged**